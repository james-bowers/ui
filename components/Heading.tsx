import * as React from "react";
import styled, { css } from "./styled";

type HeadingProps = {
  as: "h1" | "h2" | "h3" | "h4" | "h5";
  children: any;
};

const sharedHeading = css`
  font-family: ${props => props.theme.fontFamily};
`;

const H1 = styled.h1`
  font-size: 3em;
  ${sharedHeading}
`;

const H2 = styled.h2`
  font-size: 2.5em;
  ${sharedHeading}
`;

const H3 = styled.h3`
  font-size: 2em;
  ${sharedHeading}
`;

const H4 = styled.h4`
  font-size: 1.5em;
  ${sharedHeading}
`;

export default (props: HeadingProps) => {
  if (props.as === "h2") return <H2>{props.children}</H2>;
  if (props.as === "h3") return <H3>{props.children}</H3>;
  if (props.as === "h4") return <H4>{props.children}</H4>;

  return <H1>{props.children}</H1>;
};
