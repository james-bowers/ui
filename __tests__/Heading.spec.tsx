import * as React from "react";
import { configure, shallow } from "enzyme";
import * as Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";

configure({ adapter: new Adapter() });

import Heading from "../components/Heading";

describe("<Heading />", () => {
  test("renders H1", () => {
    const tree = shallow(<Heading as="h1">Heading one</Heading>);
    expect(toJson(tree)).toMatchSnapshot();
  });

  test("renders H2", () => {
    const tree = shallow(<Heading as="h2">Heading two</Heading>);
    expect(toJson(tree)).toMatchSnapshot();
  });

  test("renders H3", () => {
    const tree = shallow(<Heading as="h3">Heading two</Heading>);
    expect(toJson(tree)).toMatchSnapshot();
  });

  test("renders H4", () => {
    const tree = shallow(<Heading as="h4">Heading two</Heading>);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
