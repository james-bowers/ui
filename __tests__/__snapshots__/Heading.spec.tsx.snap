// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> renders H1 1`] = `
<styled.h1>
  Heading one
</styled.h1>
`;

exports[`<Heading /> renders H2 1`] = `
<styled.h2>
  Heading two
</styled.h2>
`;

exports[`<Heading /> renders H3 1`] = `
<styled.h3>
  Heading two
</styled.h3>
`;

exports[`<Heading /> renders H4 1`] = `
<styled.h4>
  Heading two
</styled.h4>
`;